src/lib/core/data-table/data-table.ts

```diff
@@ -76,6 +79,9 @@ export class CdkTable implements CollectionViewer {
    */
   private _columnDefinitionsByName = new Map<string,  CdkColumnDef>();
 
+  /** Differ used to find the changes in the data provided by the data source. */
+  private _dataDiffer: IterableDiffer<any> = null;
```

Comment:
One thing we should consider doing before this is released is making it `CdkTable<T>` and using the generic throughout